File: 프로그래머스 코테/프로그래머스0(61~90번)/90.접미사 배열.js
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/181909

/*
문자열 my_string이 매개변수로 주어질 때, my_string의 모든 접미사를 사전순으로 
정렬한 문자열 배열을 return 하는 solution 함수를 작성해 주세요.
*/

function solution(my_string) {
    const strLen = my_string.length
    return [...my_string].reduce((acc, cur, idx) => {
        const sliceWord = my_string.slice(idx, strLen); // 접미사 저장
        return [...acc, sliceWord];
    }, []).sort();
}

File: 프로그래머스 코테(java)/프로그래머스0(61~90번)/90.접미사 배열.java
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/181909

import java.util.Arrays;

class Solution {

    public String[] solution(String my_string) {
        String[] answer = new String[my_string.length()];

        for (int i = 0; i < answer.length; i++) {
            answer[i] = my_string.substring(i, my_string.length());
        }

        Arrays.sort(answer);
        return answer;
    }
}